ml4gw/waveforms/cbc/phenom_deco.py

```diff
@@ -0,0 +1,359 @@
+import torch
+from jaxtyping import Float
+
+from ...constants import MTSUN_SI
+from ...types import BatchTensor, FrequencySeries1d
+from .phenom_d import IMRPhenomD
+
+
+class IMRPhenomDECO(IMRPhenomD):
+    def forward(
+        self,
+        f: FrequencySeries1d,
+        chirp_mass: BatchTensor,
+        mass_ratio: BatchTensor,
+        chi1: BatchTensor,
+        chi2: BatchTensor,
+        compactness: BatchTensor,
+        distance: BatchTensor,
+        phic: BatchTensor,
+        inclination: BatchTensor,
+        f_ref: float,
+        **kwargs,
+    ):
+        """
+        IMRPhenomDECO waveform
+
+        Args:
+            f:
+                Frequency series in Hz.
+            chirp_mass:
+                Chirp mass in solar masses
+            mass_ratio:
+                Mass ratio m1/m2
+            chi1:
+                Spin of m1
+            chi2:
+                Spin of m2
+            compactness:
+                effective compactness of binary at contact
+            distance:
+                Distance to source in Mpc
+            phic:
+                Phase at coalescence
+            inclination:
+                Inclination of the source
+            f_ref:
+                Reference frequency
+
+        Returns:
+            hc, hp: Tuple[torch.Tensor, torch.Tensor]
+                Cross and plus polarizations
+        """
+        # shape assumed (n_batch, params)
+        if (
+            chirp_mass.shape[0] != mass_ratio.shape[0]
+            or mass_ratio.shape[0] != chi1.shape[0]
+            or chi1.shape[0] != chi2.shape[0]
+            or chi2.shape[0] != distance.shape[0]
+            or distance.shape[0] != phic.shape[0]
+            or phic.shape[0] != inclination.shape[0]
+        ):
+            raise RuntimeError("Tensors should have same batch size")
+        cfac = torch.cos(inclination)
+        pfac = 0.5 * (1.0 + cfac * cfac)
+
+        htilde = self.phenom_d_htilde(
+            f,
+            chirp_mass,
+            mass_ratio,
+            chi1,
+            chi2,
+            compactness,
+            distance,
+            phic,
+            f_ref,
+        )
+
+        hp = (htilde.mT * pfac).mT
+        hc = -1j * (htilde.mT * cfac).mT
+
+        return hc, hp
+
+    def phenom_d_htilde(
+        self,
+        f: FrequencySeries1d,
+        chirp_mass: BatchTensor,
+        mass_ratio: BatchTensor,
+        chi1: BatchTensor,
+        chi2: BatchTensor,
+        compactness: BatchTensor,
+        distance: BatchTensor,
+        phic: BatchTensor,
+        f_ref: float,
+    ) -> Float[FrequencySeries1d, " batch"]:
+        # PhenomDECO reuses the PhenomD phase model and the amplitude model
+        # is parametrized by an effective compactness of the two objects;
+        # see Phys. Rev. D 112, 104017 (2025) for details
+
```

Comment:
could you convert this into a docstring? It will render in the documentation page automatically.